Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,568 @@
+- `Parameter aliases` - Parameter aliases can be used in place of literal values in entity keys, function parameters, or within a $compute, $filter or $orderby expression. Example `http://host/service.svc/Employees?$filter=Region eq @p1&@p1='WA'`.
+- `Custom Query Options` - Custom query options not defined in the OData specification. They are defined by the users. They MUST NOT begin with the `$` or `@` character and MUST NOT conflict with any OData-defined system query options. Example `http://host/service/Products?debug-mode=true`.
+
+See more information on the types of query options [here](/odata/concepts/queryoptions-overview).
+
+This tutorial demonstrates how to use query options in AspNetCore OData 8.x.
+
+## Model class
+We will define our model as follows:
+
+```csharp
+public class Customer
+{
+    [Key]
+    public int Id { get; set; }
+    public String Name { get; set; }
+    public int Age { get; set; }
+    public List<Order> Orders { get; set; }
+}
+
+public class Order
+{
+    [Key]
+    public int Id { get; set; }
+    public int Price { get; set; }
+    public int Quantity { get; set; }
+}
+```
+
+We have a `Customer` entity with `Orders` navigation property. We define a one to many relationship between Customer and Order entities. One `Customer` can have one or more `Orders`. Read more on [OData data model](/odata/concepts/data-model) and [Entity relations](/odata/odatalib/edm/define-entity-relations).
+
+## Registering OData services
+
+# [.NET 6.0](#tab/net60)
+
+```csharp
+// Program.cs
+var builder = WebApplication.CreateBuilder(args);
+
+var modelBuilder = new ODataConventionModelBuilder();
+modelBuilder.EntitySet<Customer>("Customers");
+
+builder.Services.AddControllers().AddOData(
+    options => options.Select().Filter().OrderBy().Expand().Count().SetMaxTop(null).AddRouteComponents(
+        routePrefix: "odata",
+        model: modelBuilder.GetEdmModel()));
+
+var app = builder.Build();
+```
+
+# [.NET Core 3.1](#tab/netcoreapp31)
+
+```csharp
+// Startup.cs
+public void ConfigureServices(IServiceCollection services)
+{
+    var modelBuilder = new ODataConventionModelBuilder();
+    modelBuilder.EntitySet<Customer>("Customers");
+
+    services.AddControllers().AddOData(
+            options => options.Select().Filter().OrderBy().Expand().Count().SetMaxTop(null).AddRouteComponents(
+                "odata",
+                modelBuilder.GetEdmModel()));
+}
+```
+
+---
+
+In the above configuration, `options` is an instance of `ODataOptions`.
+- `Select()` enables `$select` query option.
+- `Filter()` enables `$filter` query option.
+- `OrderBy()` enables `$orderby` query option.
+- `Expand()` enables `$expand` query option.
+- `Count()` enables `$count` query option.
+- `SetMaxTop()` sets the maximum value of `$top` that a client can request. Read more in the [client-driven paging documentation](/odata/webapi-8/fundamentals/client-driven-paging)
+
```

Comment:
Might be worthwhile to mention the `EnableQueryFeatures()` shorthand method for enabling all of them at a go.